src/richie/apps/core/sitemaps.py

```diff
@@ -0,0 +1,18 @@
+"""
+Sitemap generators for contents
+"""
+from cms.sitemaps import CMSSitemap
+
+
+class CourseRunFreeSitemap(CMSSitemap):
+    """
+    A CMS pages sitemap which are free of Course run items.
+    """
+
+    def items(self):
+        titles = super().items()
+
+        # Currently a very dummy filter standing on demo behavior which create
+        # every course run with a slug starting with "sesssion". This is until
+        # how to correctly filter out course run from a cms page Title queryset.
+        return titles.exclude(slug__startswith="session")
```

Comment:
The query you're looking for should be something like:
```suggestion
        return titles.filter(page__course_run__isnull=True)
```

Reply:
Nice, i need to test this, thanks.

Reply:
Ok now we have a clean way to exclude course run and it works nice. 👌